### PR TITLE
feat(rules): RHICOMPL-2081 mark upstream rules with a new field

### PR DESCRIPTION
**Pull request by skateman**
This is the mark phase of the non-invasive cleanup of unused upstream rules from our database. I'm initially marking every rule as an upstream and during the SSG import, all the imported rules will set the `upstream` field to be `false`. This way we will know which rules should be kept in a future cleanup.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: db/migrate/20210830070917_add_upstream_for_rules.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUpstreamForRules < ActiveRecord::Migration[5.2]
+  def change
+    add_column :rules, :upstream, :boolean, null: false, default: true
+  end
```
Comment (akofink) [inline db/migrate/20210830070917_add_upstream_for_rules.rb:4]: An index may be warranted:

```suggestion
    add_column :rules, :upstream, :boolean, null: false, default: true
    add_index :rules, :upstream
  end
```

**Review by akofink — APPROVED** [review]
Thanks, @skateman!

NB: None of this code may be run due to this check which compares the number of rules: https://github.com/RedHatInsights/compliance-backend/blob/master/app/services/concerns/xccdf/benchmarks.rb#L26

It uses a strict equality check, which is flawed. If there are more rules in the benchmark saved in our DB (i.e. extra upstream rules) than the datastream XML file, it will also try re-saving any rules.

**Comment by skateman** [comment]
@akofink so what do you propose, should we get rid of this check completely?

**Comment by akofink** [comment]
> what about an ENV variable that we can set to force an import that skips this check?

Yep, my thoughts exactly :+1: No need to mess with env variables directly - just use Settings.* and env var SETTINGS__*.

**Comment by akofink** [comment]
Yes, we'll likely do a deployment with the env var set, then during the next deployment unset it again. It should be pretty straightforward. I will be happy to help.

**Review by akofink — APPROVED** [review]
Thanks, @skateman!

**Pull request merged**